lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -47,7 +47,7 @@ def client
     # @see    Transport::Base#perform_request
     #
     def perform_request(url, method, path, params={}, body=nil)
-      params = (params || {}).merge(@client_params)
+      params = (params || {}).merge(@client_params) { |key, oldval, newval| (oldval.class.to_s == 'Hash' && newval.class.to_s == 'Hash') ? oldval.merge(newval) : newval }
```

Comment:
The class check should be: `oldval.is_a?(Hash)`. This also needs an explanatory comment.